xcm/pallet-xcm-benchmarks/src/generic/mock.rs

```diff
@@ -0,0 +1,171 @@
+		XcmGenericBenchmarks: generic::{Pallet},
+	}
+);
+
+parameter_types! {
+	pub const BlockHashCount: u64 = 250;
+	pub BlockWeights: frame_system::limits::BlockWeights =
+		frame_system::limits::BlockWeights::simple_max(1024);
+}
+
+impl frame_system::Config for Test {
+	type BaseCallFilter = Everything;
+	type BlockWeights = ();
+	type BlockLength = ();
+	type DbWeight = ();
+	type Origin = Origin;
+	type Index = u64;
+	type BlockNumber = u64;
+	type Hash = H256;
+	type Call = Call;
+	type Hashing = BlakeTwo256;
+	type AccountId = u64;
+	type Lookup = IdentityLookup<Self::AccountId>;
+	type Header = Header;
+	type Event = Event;
+	type BlockHashCount = BlockHashCount;
+	type Version = ();
+	type PalletInfo = PalletInfo;
+	type AccountData = pallet_balances::AccountData<u64>;
+	type OnNewAccount = ();
+	type OnKilledAccount = ();
+	type SystemWeightInfo = ();
+	type SS58Prefix = ();
+	type OnSetCode = ();
+}
+
+/// The benchmarks in this pallet should never need an asset transactor to begin with.
+pub struct NoAssetTransactor;
+impl xcm_executor::traits::TransactAsset for NoAssetTransactor {
+	fn deposit_asset(_: &MultiAsset, _: &MultiLocation) -> Result<(), XcmError> {
+		unreachable!();
+	}
+
+	fn withdraw_asset(_: &MultiAsset, _: &MultiLocation) -> Result<Assets, XcmError> {
+		unreachable!();
+	}
+}
+
+parameter_types! {
+	pub const MaxInstructions: u32 = 100;
+}
+
+pub struct XcmConfig;
+impl xcm_executor::Config for XcmConfig {
+	type Call = Call;
+	type XcmSender = DevNull;
+	type AssetTransactor = NoAssetTransactor;
+	type OriginConverter = AlwaysSignedByDefault<Origin>;
+	type IsReserve = AllAssetLocationsPass;
+	type IsTeleporter = ();
+	type LocationInverter = xcm_builder::LocationInverter<Ancestry>;
+	type Barrier = AllowUnpaidExecutionFrom<Everything>;
+	type Weigher = xcm_builder::FixedWeightBounds<UnitWeightCost, Call, MaxInstructions>;
+	type Trader = xcm_builder::FixedRateOfFungible<WeightPrice, ()>;
+	type ResponseHandler = DevNull;
+	type AssetTrap = TestAssetTrap;
+	type AssetClaims = TestAssetTrap;
+	type SubscriptionService = TestSubscriptionService;
+}
+
+impl crate::Config for Test {
+	type XcmConfig = XcmConfig;
+	type AccountIdConverter = AccountIdConverter;
+	fn valid_destination() -> Result<MultiLocation, sp_runtime::DispatchError> {
+		let valid_destination: MultiLocation =
+			Junction::AccountId32 { network: NetworkId::Any, id: [0u8; 32] }.into();
+
+		Ok(valid_destination)
+	}
+	fn worst_case_holding() -> MultiAssets {
+		crate::mock_worst_case_holding()
+	}
+}
+
+impl generic::Config for Test {
+	type Call = Call;
+
+	fn worst_case_response() -> (u64, Response) {
+		let assets: MultiAssets = (Concrete(Here.into()), 100).into();
+		(0, Response::Assets(assets))
+	}
+
+	fn transact_origin() -> Option<MultiLocation> {
+		Some(Default::default())
+	}
+
+	fn subscribe_origin() -> Option<MultiLocation> {
+		Some(Default::default())
+	}
+
+	fn claimable_asset() -> Option<(MultiLocation, MultiLocation, MultiAssets)> {
```

Comment:
Why is this not returning `BenchmarkError` directly? You usually do `.map_err(|_| BenchmarkError::Skip)` inside the benchmark code. This seems like a decision that should be made by the top level runtime.

Reply:
As in, for some cases I might want to do `Override` and for other `Skip`.

Reply:
:+1: If that's possible I really like that idea.

Reply:
yup, great suggestion